gc/base/standard/Scavenger.cpp

```diff
@@ -3938,12 +3958,31 @@ MM_Scavenger::fixupSlot(GC_SlotObject *slotObject)
 	return false;
 }
 
+bool
+MM_Scavenger::fixupSlot(omrobjectptr_t *slotPtr)
+{
+	omrobjectptr_t objectPtr = *slotPtr;
+	bool const compressed = _extensions->compressObjectReferences();
+	if (NULL != objectPtr) {
+		MM_ForwardedHeader forwardHeader(objectPtr, compressed);
+		if (forwardHeader.isStrictlyForwardedPointer()) {
+			*slotPtr = forwardHeader.getForwardedObject();
+			Assert_MM_false(isObjectInEvacuateMemory(*slotPtr));
+			return true;
+		} else {
+			Assert_MM_false(_extensions->objectModel.isDeadObject(objectPtr));
+		}
+	}
+	return false;
+}
+
```

Comment:
can we share the code by rewriting the original method to use this one, and move it to hpp and inline it?

```
bool
MM_Scavenger::fixupSlot(GC_SlotObject *slotObject)
{
	omrobjectptr_t objectPtr = slotObject->readReferenceFromSlot();
	if (fixupSlot(&objectPtr)) {
		slotObject->writeReferenceToSlot(objectPtr);
		return true;
	}
	return false;
}
```